community/detectors/joomla_cve_2023_23752/src/test/java/com/google/tsunami/plugins/detectors/cves/cve202323752/Cve202323752VulnDetectorTest.java

```diff
@@ -0,0 +1,192 @@
+
+  @Inject private Cve202323752VulnDetector detector;
+
+  private final MockWebServer mockWebServer = new MockWebServer();
+
+  private NetworkService joomlaService;
+  private static final String LeakedDataJsonSample =
+      "{\n"
+          + "    \"data\": [\n"
+          + "        {\n"
+          + "            \"type\": \"application\",\n"
+          + "            \"id\": \"224\",\n"
+          + "            \"attributes\": {\n"
+          + "                \"host\": \"google.com\",\n"
+          + "                \"id\": 224\n"
+          + "            }\n"
+          + "        },\n"
+          + "        {\n"
+          + "            \"type\": \"application\",\n"
+          + "            \"id\": \"224\",\n"
+          + "            \"attributes\": {\n"
+          + "                \"user\": \"root\",\n"
+          + "                \"id\": 224\n"
+          + "            }\n"
+          + "        },\n"
+          + "        {\n"
+          + "            \"type\": \"application\",\n"
+          + "            \"id\": \"224\",\n"
+          + "            \"attributes\": {\n"
+          + "                \"password\": \"example\",\n"
+          + "                \"id\": 224\n"
+          + "            }\n"
+          + "        }\n"
+          + "    ]\n"
+          + "}";
+  private TargetInfo targetInfo;
+
+  @Before
+  public void setUp() throws IOException {
+    mockWebServer.start();
+    Guice.createInjector(
+            new FakeUtcClockModule(fakeUtcClock),
+            new HttpClientModule.Builder().build(),
+            new Cve202323752DetectorBootstrapModule())
+        .injectMembers(this);
+
+    joomlaService =
+        NetworkService.newBuilder()
+            .setNetworkEndpoint(
+                forHostnameAndPort(mockWebServer.getHostName(), mockWebServer.getPort()))
+            .setTransportProtocol(TransportProtocol.TCP)
+            .setSoftware(Software.newBuilder().setName("joomla 4.2.6-php8.0"))
+            .setServiceName("http")
+            .build();
+
+    targetInfo =
+        TargetInfo.newBuilder()
+            .addNetworkEndpoints(
+                forHostnameAndPort(mockWebServer.getHostName(), mockWebServer.getPort()))
+            .build();
+  }
+
+  @After
+  public void tearDown() throws IOException {
+    mockWebServer.shutdown();
+  }
+
+  final Dispatcher dispatcher =
+      new Dispatcher() {
+
+        @Override
+        public MockResponse dispatch(RecordedRequest request) throws InterruptedException {
+          if (request.getPath().equals("/api/index.php/v1/config/application?public=true")) {
+            return new MockResponse()
+                .addHeader("Content-Type", "application/json; charset=utf-8")
+                .setBody(LeakedDataJsonSample)
+                .setResponseCode(HttpStatus.OK.code());
+          }
+          return new MockResponse().setResponseCode(404);
+        }
+      };
+
+  @Test
+  public void DetectCVE202323752() {
```

Comment:
```suggestion
  public void detect_whenVulnerable_returnsVulnerability() {
```
Follow style for test names